controller/utils/fio-chain.js

```diff
@@ -37,146 +111,193 @@ export const getLastIrreversibleBlockOnFioChain = async () => {
 
-const getActions = async (accountName, pos, offset) => {
-  let actionsHistoryResponse;
-
-  // Schedule the request based on the rate limiter
-  await rateLimiterFor1000Rpm.scheduleRequest();
-
+export const getTableRows = async ({ logPrefix, tableRowsParams }) => {
   try {
-    actionsHistoryResponse = await fetch(
-      FIO_SERVER_URL_HISTORY + 'v1/history/get_actions',
-      {
-        body: JSON.stringify({ account_name: accountName, pos, offset }),
+    const tableRowsResponse = await fetchWithRateLimit({
+      url: makeTableRowsUrl(FIO_SERVER_URL_ACTION),
+      options: {
+        body: JSON.stringify(tableRowsParams),
         method: 'POST',
       },
-    );
-
-    if (actionsHistoryResponse.status === 429) {
-      console.log('Rate limit exceeded (429), waiting for 60 seconds...');
-      await sleep(MINUTE_IN_MILLISECONDS); // Wait for 60 seconds before retrying
-      return await getActions(accountName, pos, offset); // Retry the request
-    }
+      backupUrl: FIO_SERVER_URL_ACTION_BACKUP
+        ? makeTableRowsUrl(FIO_SERVER_URL_ACTION_BACKUP)
+        : null,
+    });
 
-    await checkHttpResponseStatus(
-      actionsHistoryResponse,
-      'Getting FIO actions history went wrong.',
-    );
-  } catch (e) {
-    console.error(e);
-    if (FIO_SERVER_URL_HISTORY_BACKUP) {
-      actionsHistoryResponse = await fetch(
-        FIO_SERVER_URL_HISTORY_BACKUP + 'v1/history/get_actions',
-        {
-          body: JSON.stringify({ account_name: accountName, pos, offset }),
-          method: 'POST',
-        },
-      );
-
-      if (actionsHistoryResponse.status === 429) {
-        console.log('Rate limit exceeded (429), waiting for 60 seconds...');
-        await sleep(MINUTE_IN_MILLISECONDS); // Wait for 60 seconds before retrying
-        return await getActions(accountName, pos, offset); // Retry the request
-      }
-
-      await checkHttpResponseStatus(
-        actionsHistoryResponse,
-        'Getting FIO actions history went wrong.',
-      );
-    }
+    return tableRowsResponse ? tableRowsResponse.json() : null;
+  } catch (error) {
+    handleChainError({
+      logMessage: `${logPrefix} ${error}`,
+      consoleMessage: `GET TABLE ROWS ERROR: ${error}`,
+    });
   }
-  const actionsHistory = actionsHistoryResponse
-    ? await actionsHistoryResponse.json()
-    : null;
+};
+
+export const getLastFioOracleItemId = async () => {
+  const logPrefix = 'Get last FIO Oracle Item ID';
+  try {
+    const tableRowsParams = {
+      json: true,
+      code: FIO_ACCOUNT_NAMES.FIO_ORACLE,
+      scope: FIO_ACCOUNT_NAMES.FIO_ORACLE,
+      table: FIO_TABLE_NAMES.FIO_ORACLE_LDGRS,
+      limit: 1,
+      reverse: true,
+    };
 
-  return actionsHistory;
+    const response = await getTableRows({
+      logPrefix,
+      tableRowsParams,
+    });
+
+    return response && response.rows[0] ? response.rows[0].id + 1 : null; // Set id + 1 because if we will set id as it is then it will process already processed wrap item
```

Comment:
does it always exist? `response.rows[0].id + 1`

Reply:
and should weforce number type for `response.rows[0].id`

Reply:
Done